providers/default/defltprov.c

```diff
@@ -272,21 +271,21 @@ static const OSSL_ALGORITHM deflt_kdfs[] = {
     { OSSL_KDF_NAME_X942KDF, "default=yes", kdf_x942_kdf_functions },
 #endif
 #ifndef OPENSSL_NO_SCRYPT
-    { OSSL_KDF_NAME_SCRYPT, "default=yes", kdf_scrypt_functions },
+    { "SCRYPT:id-scrypt", "default=yes", kdf_scrypt_functions },
 #endif
    { NULL, NULL, NULL }
 };
 
 static const OSSL_ALGORITHM deflt_keyexch[] = {
 #ifndef OPENSSL_NO_DH
-    { "dhKeyAgreement", "default=yes", dh_keyexch_functions },
+    { "dhKeyAgreement:DH", "default=yes", dh_keyexch_functions },
 #endif
     { NULL, NULL, NULL }
 };
 
 static const OSSL_ALGORITHM deflt_signature[] = {
 #ifndef OPENSSL_NO_DSA
-    { "DSA", "default=yes", dsa_signature_functions },
+    { "dsaEncryption:DSA", "default=yes", dsa_signature_functions },
```

Comment:
I know it makes no difference technically. But I'd suggest we adopt a convention of having a preferred "canonical" name as being the first one in the list. It would be this name that we refer to throughout the documentation for example.

Reply:
Considering we used to have NIDs all over the place, it's not at all easy to know what the "canonical" name is.  But as long as we agree, I'm game.
(I personally don't care, precisely because it doesn't matter technically)

Reply:
I think we ignore any "old" names that we have and pick a naming scheme for our "friendly" names and use that. If it happens to coincide with the any of the old names, then great.

Reply:
There's a problem with that...  if someone uses something like `EVP_PKEY_DSA`, which is an alias for a NID, and someone else uses either `OBJ_nid2sn` or `OBJ_nid2ln` on it an uses that name in a fetch, there are going to be some misses unless we make sure that our providers give the full set of names.  Incidently, this is something we need to be careful with any time we do implicit fetches.

Reply:
I think you misunderstand my suggestion. I only meant "ignore any old names" in terms of picking our friendly name. Choose a scheme for our friendly names forgetting about what we've called things before. Make that the first listed name. After that list all the aliases that we've historically called an algorithm by. We should encourage use of the "friendly name" from hereon in.

Reply:
Ah, right, that makes a lot more sense.

So now, we only have to select the friendly name.  Wanna do the honors and go through the list and suggest which one should be the friendly name (I really don't mind if you do that as bulk suggestion where applicable)

Reply:
My suggestion for a naming scheme:

[ALGNAME][VERSION?][-SUBNAME[VERSION?]?][-SIZE?][-MODE?]

Although we're case insensitive, we should still present all names in caps in the "canonical" form.
Subname may be present where we are combining multiple algorithms together, e.g. MD5-SHA1
Version is only present if there are multiple versions of an alg (MD2, MD4, MD5). It may be omitted if there is only one version (if in the future a subsequent version is released we can always change the canonical name, and add the old name as an alias)
Size is only present if multiple versions of an algorithm exist with different sizes (e.g. AES-128-CBC, AES-256-CBC)
Mode is only present where applicable.

Examples:
SHA1
SHA2-256
SHA3-224
BLAKE2S-256
KECCAK-KMAC-128
ARIA-256-CBC
DH
DSA

Reply:
Cool, will do

Reply:
Ho boy, naming can be fun...  I was wondering about "SHA2-256", 'cause I couldn't recognise it, and no wonder, I actually think you just invented *yet another* set of names for the SHA2 family.  NIST uses SHA-1, SHA-224, SHA-256,SHA-384, SHA-512, SHA-512/224 and SHA-512/256 (see https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf)

So question is, if we want some *real world* names, must we do "SHA2-xxx"?

Reply:
The SHA2's would be algname-size wouldn't they? (i.e. conforming to the naming scheme)
SHA-1 is the odd one out.

Reply:
Well we have SHA1 and SHA3, so it seems odd that SHA2 doesn't have the version number included. Personally I'm happy with calling it SHA2-256 etc, with SHA-256 as an alias. It makes it clearer - even though it doesn't entirely conform to NIST naming. But if that is considered a step too far then we can modify my suggested naming scheme to include some additional weasel words about when the version number can be optional. E.g.
 
"Version is only present if there are multiple versions of an alg (MD2, MD4, MD5). It may be omitted if there is only one version (if in the future a subsequent version is released we can always change the canonical name, and add the old name as an alias), or if common practice for a particular version is to omit it"

Reply:
Using SHA2-256 seems sensible, with SHA-256 as an alias.

Reply:
I can go for SHA2-nnn, no problem.  I'm just amused that we end up inventing new names for this. :wink:

Reply:
Done